packages/server/billing/helpers/generateInvoice.ts

```diff
@@ -339,7 +350,7 @@ export default async function generateInvoice(
   const isUpcoming = type === 'upcoming'
 
   let status: InvoiceStatusEnum = isUpcoming ? 'UPCOMING' : 'PENDING'
-  if (status === 'PENDING' && invoice.closed === true) {
```

Comment:
The `invoice` object no longer has the `closed` field. I believe our subscriptions always charge the card automatically, although there may be exceptions with Enterprise? If subscriptions are always charged automatically, the invoice is `PENDING`, but not paid, I think it's safe to assume it's a `FAILED` payment.

Stripe has created [invoice statuses](https://stripe.com/docs/invoicing/overview#invoice-statuses), so we could refactor our [InvoiceStatusEnum](https://github.com/ParabolInc/parabol/blob/dd8e82c239be9526a5ee6f443fea4b9aeff51e2d/packages/server/database/types/Invoice.ts#L6) in a separate PR to use this?

Reply:
IIRC after stripe finalizes an invoice, it waits an hour or 2 until it creates a charge.
During that timeframe would invoices look like they're failed?

Reply:
Interesting! I'm now fetching the `charge` and checking the status there